packages/cogify/src/cogify/stac.ts

```diff
@@ -62,37 +63,88 @@ export interface CogifyCreationOptions {
   background?: Rgba;
 }
 
+export interface GeneratedProperties {
+  /** Package name that generated the file */
+  package: string;
+
+  /** Version number that generated the file */
+  version: string;
+
+  /** Git commit hash that the file was generated with */
+  hash: string;
+
+  /** ISO date of the time this file was generated */
+  datetime: string;
+
+  /** version of GDAL used to create the COG */
+  gdal?: string;
+
+  /**
+   * is the tiff invalid
+   *
+   * If the tiff was generated but then determined to be invalid this property explains why the tiff was rejected.
+   *
+   * Reasons:
+   * - "empty" - No data was produced by GDAL when creating the tiff
+   */
+  invalid?: 'empty';
+}
+
 export type GdalResampling = 'nearest' | 'bilinear' | 'cubic' | 'cubicspline' | 'lanczos' | 'average' | 'mode';
 
 export type CogifyStacCollection = StacCollection;
+
 export type CogifyStacItem = StacItem & {
   properties: {
-    'linz_basemaps:generated': {
-      /** Package name that generated the file */
-      package: string;
-      /** Version number that generated the file */
-      version: string;
-      /** Git commit hash that the file was generated with */
-      hash: string;
-      /** ISO date of the time this file was generated */
-      datetime: string;
-      /** version of GDAL used to create the COG */
-      gdal?: string;
-
-      /**
-       * is the tiff invalid
-       *
-       * If the tiff was generated but then determined to be invalid this property explains why the tiff was rejected.
-       *
-       * Reasons:
-       * - "empty" - No data was produced by GDAL when creating the tiff
-       */
-      invalid?: 'empty';
-    };
+    'linz_basemaps:generated': GeneratedProperties;
     'linz_basemaps:options': CogifyCreationOptions;
   };
 };
 
+export type TopoStacItem = StacItem & {
+  properties: {
+    /**
+     * A topo raster map sheet's code.
+     *
+     * @example "CJ10"
+     */
+    map_code: string;
+
+    /**
+     * A map sheet's version.
+     *
+     * @example "v1.00"
+     */
+    version: string;
+
+    /**
+     * An EpsgCode Enum representing a map sheet's projection.
+     *
+     * @example EpsgCode.Nztm2000 = 2193
+     */
+    'proj:epsg': EpsgCode;
+
+    /**
+     * The width of a map sheet in pixels.
+     */
+    'source.width': number;
```

Comment:
`.` inside json objects property names can cause some tools to have lots of problems, some tools use it to expect object nesting `{ source: { width: 100 } }` could these be moved or renamed?

We should also be prefixing keys that we are adding eg `linz_basemaps:map_code`